src/simplify_algebra.cpp

```diff
@@ -1393,16 +1398,39 @@ struct find_split_reshape
         {
             rsp_axis = std::distance(rsp_strides.begin(), ait);
         }
-        // calculate reshape output shape
-        std::vector<int64_t> vec_dims(vec_rsp.size());
-
-        std::transform(vec_rsp.begin(), vec_rsp.end(), vec_dims.begin(), [&](auto is) {
-            return is->get_shape().lens()[rsp_axis];
-        });
 
+        // calculate reshape output shape
         std::vector<int64_t> rsp_out_lens(rsp_lens.begin(), rsp_lens.end());
+        rsp_out_lens[rsp_axis] = 1;
+        auto rsp_fixed_size    = std::accumulate(
+            rsp_out_lens.begin(), rsp_out_lens.end(), 1, std::multiplies<std::size_t>());
+        // cannot create a valid reshape for simplification
+        if(input_size % rsp_fixed_size != 0)
+        {
+            return;
+        }
+        auto rsp_axis_len      = input_size / rsp_fixed_size;
+        rsp_out_lens[rsp_axis] = rsp_axis_len;
+
+        // calculate new slice start and end indices
+        // indices are scaled using the new reshape axis and the original slice axis
+        // eg. slice with size [1, 2, 4, 30] can be reshaped to [8, 30] and vice-versa
+
+        std::vector<int64_t> new_starts(vec_rsp.size());
+        std::transform(vec_rsp.begin(), vec_rsp.end(), new_starts.begin(), [&](auto is) {
+            auto cont   = is->inputs().front();
+            auto og_slc = cont->inputs().front();
+            return any_cast<op::slice>(og_slc->get_operator()).starts[0] * rsp_axis_len /
+                   slc_axis_len;
```

Comment:
Is the `start` guaranteed to result in integer?

Reply:
The only other possibility as far as I understand is you can technically have an empty slice op (axes={}, starts={}, ends={}) which would be problematic. But that should never happen because that sort of instruction gets eliminated as dead code. And if it somehow makes it to this matcher, `get_splits` will be empty and we would return right away. 
Other than that, starts is defined as std::vector<int64_t>